src/valkey-cli.c

```diff
@@ -2226,9 +2228,35 @@ static int cliReadReply(int output_raw_strings) {
         fflush(stdout);
         sdsfree(out);
     }
+    
     return REDIS_OK;
 }
 
+/* Helper method to process unsubscribe responses */
+static void handlePubSubMode(redisReply *reply) {
+    if (reply->elements >= 3) {
+        char *cmd = reply->element[0]->str;
+        int count = reply->element[2]->integer;
+ 
+        /* Update counts based on the command type */
+        if (strcmp(cmd, "subscribe") == 0 || strcmp(cmd, "psubscribe") == 0 || 
+            strcmp(cmd, "unsubscribe") == 0 || strcmp(cmd, "punsubscribe") == 0) {
+            config.pubsub_unsharded_count = count;
+        } else if (strcmp(cmd, "ssubscribe") == 0 || strcmp(cmd, "sunsubscribe") == 0) {
+            config.pubsub_sharded_count = count;
+        }
+
+        /* Update pubsub mode based on the current counts */
+        if (config.pubsub_unsharded_count == 0 && config.pubsub_sharded_count == 0 && config.pubsub_mode) {
```

Comment:
nit: we could make this condition symmetrical to your approach in the below condition 
```suggestion
        if (config.pubsub_unsharded_count + config.pubsub_sharded_count == 0 && config.pubsub_mode) {
```